code/modules/clothing/suits/marine_armor.dm

```diff
@@ -174,18 +172,45 @@ var/list/squad_colors_chat = list(rgb(230,125,125), rgb(255,230,80), rgb(255,150
 		turn_off_light(user)
 	..()
 
+/obj/item/clothing/suit/storage/marine/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/device/flashlight/shoulder))
+		if(flags_marine_armor & ARMOR_LAMP_OVERLAY)
+			to_chat(user, SPAN_WARNING("\The [src] already has a TNR shoulder lamp attached."))
+			return
+		if(!(initial(flags_marine_armor) & ARMOR_LAMP_OVERLAY))
+			to_chat(user, SPAN_WARNING("\The [src] doesn't have an attachment point for a TNR shoulder lamp."))
+			return
+		var/obj/item/device/flashlight/shoulder/shoulder_light = W
+		flags_marine_armor |= ARMOR_LAMP_OVERLAY
+		var/datum/action/item_action/toggle/flash_toggle = new(src)
+		LAZYADD(actions, flash_toggle)
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			if(H.wear_suit == src)
+				flash_toggle.give_to(usr)
+		verbs += /obj/item/clothing/suit/storage/marine/proc/remove_armor_light
+		if(shoulder_light.on)
+			turn_on_light(user)
+		to_chat(user, SPAN_WARNING("You attach \the [shoulder_light] to \the [src]."))
+		update_icon(user)
+		qdel(shoulder_light)
```

Comment:
handle this via forcemoving into contents instead of via new and qdel()

this makes adding different kinds of shoulder lamp thingy easier in the future